extractor/image/image.go

```diff
@@ -93,3 +109,33 @@ func fileOpener(fileName string) func() (io.ReadCloser, error) {
 		return ioutil.NopCloser(r), nil
 	}
 }
+
+func tryOCI(fileName string) (v1.Image, error) {
+	lp, err := layout.FromPath(fileName)
+	if err != nil {
+		return nil, xerrors.Errorf("unable to open %s as an OCI image: %w", fileName, err)
+	}
+
+	index, err := lp.ImageIndex()
+	if err != nil {
+		return nil, xerrors.Errorf("unable to retrieve index.json: %w", err)
+	}
+
+	m, err := index.IndexManifest()
+	if err != nil {
+		return nil, xerrors.Errorf("invalid index.json: %w", err)
+	}
+
+	if len(m.Manifests) == 0 {
+		return nil, xerrors.New("no valid manifest")
+	}
+
+	// Support only first image
+	h := m.Manifests[0].Digest
```

Comment:
If we only support first image, does this still make us say we're OCI compliant?

Reply:
`containers/image` does the same thing if any tag/digests are not specified, but they also support multiple images.
https://github.com/containers/image/blob/d13a90f787a7d3cd58e6f13c7cd1f2307b74a1ef/oci/layout/oci_transport.go#L188-L194

If we support multiple images, it makes the implementation a bit complicated. A user needs to specify which image is scanned by digest like the following.

```
$ fanal -f /tmp/alpine.oci@sha256:24fbbcafb45a7030f7d53035753733950ec23320cecb4d10d0f1d2f1da9c62f0
```

This is because a tag like `latest` is optional for OCI, so we need a digest. IMO, there is only 1 image in OCI directory in many cases.

```
// select the first image
$ fanal -f /tmp/alpine.oci
```

I feel it is enough at the moment. If someone raises an issue, we can support multiple images.

Reply:
OK! I was just curious.